src/lib/y2partitioner/widgets/overview.rb

```diff
@@ -0,0 +1,246 @@
+      def contents
+        Tree(Id(:tree), Opt(:notify), _("System View"), items)
+      end
+
+      # handles widgets. As it is with notify, it will get any click on Item
+      def handle(event)
+        id = event["ID"]
+        log.info "handling id #{id}"
+
+        nil
+      end
+
+    private
+
+      def items
+        [
+          Item(
+            Id(:all),
+            term(:icon, Icons::ALL),
+            open?(:all),
+            "machine", # TODO: stuck getting hostname on my pc Yast::Hostname.CurrentHostname,
+            machine_items
+          ),
+          # TODO: only if there is graph support UI.HasSpecialWidget(:Graph)
+          device_graph_item,
+          mount_graph_item,
+          summary_item,
+          settings_item
+        ]
+      end
+
+      def device_graph_item
+        Item(
+          Id(:devicegraph),
+          term(:icon, Icons::GRAPH),
+          _("Device Graph"),
+          open?(:devicegraph)
+        )
+      end
+
+      def mount_graph_item
+        Item(
+          Id(:mountgraph),
+          term(:icon, Icons::GRAPH),
+          _("Mount Graph"),
+          open?(:mountgraph)
+        )
+      end
+
+      def summary_item
+        Item(
+          Id(:summary),
+          term(:icon, Icons::SUMMARY),
+          _("Installation Summary"),
+          open?(:summary)
+        )
+      end
+
+      def settings_item
+        Item(
+          Id(:settings),
+          term(:icon, Icons::SETTINGS),
+          _("Settings"),
+          open?(:settings)
+        )
+      end
+
+      def machine_items
+        [
+          harddisk_items,
+          raid_items,
+          lvm_items,
+          crypt_files_items,
+          device_mapper_items,
+          nfs_items,
+          btrfs_items,
+          tmpfs_items,
+          unused_items
+        ]
+      end
+
+      def harddisk_items
+        Item(
+          Id(:hd),
+          term(:icon, Icons::HD),
+          _("Hard Disks"),
+          open?(:hd),
+          disks_items
+        )
+      end
+
+      def disks_items
+        @device_graph.disks.map do |disk|
+          Item(
+            Id(disk.name),
+            disk.sysfs_name,
+            open?(disk.name),
+            partition_items(disk)
+          )
+        end
+      end
+
+      def partition_items(disk)
+        disk.partitions.map do |partition|
+          Item(
+            Id(partition.name),
+            partition.sysfs_name,
+            open?(partition.name)
+          )
+        end
+      end
+
+      def raid_items
+        Item(
+          Id(:raid),
+          term(:icon, Icons::RAID),
+          _("RAID"),
+          open?(:raid),
+          [] # TODO: real MD subtree
+        )
+      end
+
+      def lvm_items
+        Item(
+          Id(:lvm),
+          term(:icon, Icons::LVM),
+          _("Volume Management"),
+          open?(:lvm),
+          lvm_vgs_items
+        )
+      end
+
+      def lvm_vgs_items
+        @device_graph.lvm_vgs.map do |vg|
```

Comment:
NP: Sandy Metz would advise you to use a private attr_reader for device_graph instead of accessing `@device_graph` directly.

BTW, since the class is called `Devicegraph` (a single word, although I agree it should have been called DeviceGraph), I find the underscore strange. But that's just a matter of taste.